kolibri/plugins/learn/assets/src/vue/search-widget/index.vue

```diff
@@ -0,0 +1,327 @@
+    },
+  };
+
+</script>
+
+
+<style lang="stylus" scoped>
+
+  @require '~core-theme.styl'
+  $thumbnail-width = 200
+  $margin-width = 24
+  card-list-container-width(n-cols)
+    width: ($thumbnail-width * n-cols) + ($margin-width * n-cols)
+  card-list-container-max-width(n-cols)
+    max-width: ($thumbnail-width * n-cols) + ($margin-width * n-cols)
+  input-width(n-cols)
+    width: $thumbnail-width * (n-cols - 1)
+  media-query(n-cols)
+    .page-container
+      max-width: ($thumbnail-width * n-cols) + $margin-width * (n-cols - 1)
+    .card-list-container
+      max-width: ($thumbnail-width * n-cols) + $margin-width * n-cols
+    .tool-bar-container
+      width: ($thumbnail-width * n-cols) + $margin-width * (n-cols - 1)
+    .tool-bar-container .search-input-active
+      width: $thumbnail-width * (n-cols) * 0.7
+    .tool-bar-container .tool-bar-center
+      -webkit-transform: translateX(-(((($thumbnail-width * n-cols) + $margin-width * (n-cols - 1)) - ($thumbnail-width * (n-cols) * 0.7))/2)px)
+  .section-title
+    margin-top: 5vh
+    font-size: 1.2em
+    font-weight: 700
+  .tool-bar-container
+    position: fixed
+    top: 0
+    card-list-container-width(1)
+    height: 30px
+    padding: 30 0
+    background-color: $core-bg-canvas
+    text-align: center
+    z-index: 1
+    .breadcrumbs-container
+      position: absolute
+      left:0
+      display: inline-block
+      line-height: 30px
+      z-index: -1000
+    .tool-bar
+      float: right
+      display: inline-block
+      opacity: 0.6
+      transition: all 0.3s ease-out
+    .tool-bar-center
+      height: 30px
+      text-align: center
+    .tool-bar-center .btn-search
+      pointer-events: none
+      top: 0
+    select.btn-channel
+      display: inline-block
+      position: absolute
+      right: 50px
+      width: 160px
+      padding: 0.2em 0.8em
+      background: url('./arrow-down.svg') no-repeat right
+      border: 1px solid $core-text-default
+      border-radius: 50px
+      -webkit-appearance: none
+    .searchform
+      display: inline-block
+      z-index: 1
+    .search-input
+      outline: none
+      position: relative
+      display: inline-block
+      width:0
+      background-color: $core-bg-canvas
+      border-radius: 40px
+      border: 1px solid rgba(58, 58, 58, 0.1)
+      border:none
+      pointer-events: none
+      transition: width 0.2s ease-out
+      -webkit-backface-visibility: hidden
+      -webkit-transform: translate3d(0, 0, 0)
+    .search-input-active
+      input-width(1)
+      display: inline-block
+      padding: 0 40px
+      height:30px
+      border: 1px solid rgba(58, 58, 58, 1)
+      pointer-events: auto
+    .btn-search-left
+      left: 40px
+    .btn-search
+      position: relative
+      display: inline-block
+      height: 30px
+      width: 30px
+      background:none
+      border: none
+      text-indent: -10000px
+      cursor: pointer
+      z-index: 1
+      top: 4
+      .btn-search-img
```

Comment:
We should avoid nesting selectors whenever possible. Since the styles are scoped there is no danger of class collisions, and nesting comes with a bunch of disadvantages.

For more info:
* http://csswizardry.com/2012/05/keep-your-css-selectors-short/
* http://thesassway.com/beginner/the-inception-rule